b3lb/rest/classes/api.py

```diff
@@ -476,22 +468,18 @@ def get_secret_metrics(self) -> str:
         return SecretMetricsList.objects.get(secret=self.secret).metrics
 
     @staticmethod
-    def get_sha_algorithm(sha: str) -> Any:
-        if sha not in settings.B3LB_ALLOWED_SHA_ALGORITHMS or sha not in SHA_ALGORITHMS_BY_STRING:
+    def get_sha_algorithm(sha: str) -> Union[HASH, None]:
+        if sha not in settings.B3LB_ALLOWED_SHA_ALGORITHMS or sha not in Cluster.SHA_BY_STRING:
             return None
-        return SHA_ALGORITHMS_BY_STRING[sha]
-
-    def get_sha_algorithm_by_cluster_group(self) -> Any:
-        cluster_group = ClusterGroupRelation.objects.filter(cluster=self.node.cluster)
-        return self.get_sha_algorithm(cluster_group[0].cluster_group.sha_function)
+        return Cluster.SHA_BY_STRING.get(sha)()
 
-    def get_sha_algorithm_by_checksum_length(self) -> Any:
+    def get_sha_algorithm_by_checksum_length(self) -> Union[HASH, None]:
         check_length = len(self.checksum)
-        if check_length not in SHA_ALGORITHMS_BY_LENGTH:
+        if check_length not in Cluster.SHA_BY_LENGTH:
```

Comment:
Wouldn't be a `Cluster.get(...)` more efficient? Use `None` instead of a arbitrary string value?

Reply:
it's now more efficient